Analytics/Request/BlockingRequestHandler.cs

```diff
@@ -131,41 +132,88 @@ internal BlockingRequestHandler(Client client, TimeSpan timeout)
 					{ "batch size", batch.batch.Count }
 				});
 
+				const int MAXIMUM_BACKOFF_DURATION = 10000;
+				int backoff = 100;
+
+				int statusCode = (int)HttpStatusCode.OK;
+				string responseStr = "";
+
+				while (backoff < MAXIMUM_BACKOFF_DURATION)
+				{
 #if NET35
-                watch.Start();
-
-                try
-                {
-                    var response = Encoding.UTF8.GetString(_httpClient.UploadData(uri, "POST", Encoding.UTF8.GetBytes(json)));
-                    watch.Stop();
-
-                    Succeed(batch, watch.ElapsedMilliseconds);
-                }
-                catch (WebException ex)
-                {
-                    watch.Stop();
-
-                    string responseStr = ex.Message;
-                    Fail(batch, new APIException("Unexpected Response", responseStr), watch.ElapsedMilliseconds);
-                }
+					watch.Start();
+
+					try
+					{
+						var response = Encoding.UTF8.GetString(_httpClient.UploadData(uri, "POST", Encoding.UTF8.GetBytes(json)));
+						watch.Stop();
+
+						Succeed(batch, watch.ElapsedMilliseconds);
+						break;
+					}
+					catch (WebException ex)
+					{
+						watch.Stop();
+
+						var response = (HttpWebResponse)ex.Response;
+						if (response != null)
+						{
+							statusCode = (int)response.StatusCode;
+							if (statusCode >= 500 || statusCode == 429)
```

Comment:
Let's make this condition: `if ((statusCode >= 500 && statusCode <= 600) || statusCode == 429)`. Technically there's a HTTP 6xx class of codes.